lib/pages/scout/dialogs/RedeemDialog.dart

```diff
@@ -386,13 +386,15 @@ class _RedeemDialogState extends State<RedeemDialog> {
                           1,
                           Colors.transparent),
                       child: TextButton(
-                        onPressed: () {
-                          lspController.redeem();
-                          Navigator.pop(context);
-                          showDialog(
-                              context: context,
-                              builder: (BuildContext context) =>
-                                  confirmTransaction(context, true, ""));
+                        onPressed: () async {
+                          final result = await lspController.redeem();
+
+                          if (result) {
+                            showDialog(
+                                context: context,
+                                builder: (BuildContext context) =>
+                                    confirmTransaction(context, true, ""));
+                          }
```

Comment:
For better UX we should have a dialog for when the user rejects the transaction. It should notify the user that the transaction has not gone through or something like that.

Reply:
okay understood